scripts/zkai_attention_kv_d64_four_head_seq32_air_private_softmax_table_lookup_gate.py

```
@@ -0,0 +1,919 @@
+    except OSError as err:
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(f"failed to read {label}: {err}") from err
+    if len(raw) != expected_size:
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(
+            f"{label} read size drift: stat={expected_size}, read={len(raw)}"
+        )
+    return raw
+
+
+def bounded_file_size(path: pathlib.Path, max_bytes: int, label: str) -> int:
+    if not path.is_file():
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(f"missing {label}: {path}")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)